extension/sqlpgq/sqlpgq_functions/sqlpgq_scan.cpp

```diff
@@ -0,0 +1,300 @@
+namespace duckdb {
+
+static void ScanCSREFunction(ClientContext &context, TableFunctionInput &data_p, DataChunk &output) {
+	bool &gstate = ((CSRScanState &)*data_p.global_state).finished;
+
+	if(gstate) {
+                output.SetCardinality(0);
+                return;
+        }
+
+	gstate = true;
+
+        auto sqlpgq_state_entry = context.registered_state.find("sqlpgq");
+        if (sqlpgq_state_entry == context.registered_state.end()) {
+                //! Wondering how you can get here if the extension wasn't loaded, but leaving this check in anyways
+                throw InternalException("The SQL/PGQ extension has not been loaded");
+        }
+        auto sqlpgq_state = reinterpret_cast<SQLPGQContext *>(sqlpgq_state_entry->second.get());
+
+	auto csrID = ((CSRScanEData *)(data_p.bind_data))->csrID;
+        CSR *csr = sqlpgq_state -> GetCSR(csrID);
+        output.SetCardinality(csr->e.size());
+        output.data[0].SetVectorType(VectorType::FLAT_VECTOR);
+        FlatVector::SetData(output.data[0], (data_ptr_t)csr->e.data());
+}
+
+static void ScanCSRVFunction(ClientContext &context, TableFunctionInput &data_p, DataChunk &output) {
+        bool &gstate = ((CSRScanState &)*data_p.global_state).finished;
+
+        if(gstate) {
+                output.SetCardinality(0);
+                return;
+        }
+
+        gstate = true;
+
+        auto sqlpgq_state_entry = context.registered_state.find("sqlpgq");
+        if (sqlpgq_state_entry == context.registered_state.end()) {
+                //! Wondering how you can get here if the extension wasn't loaded, but leaving this check in anyways
+                throw InternalException("The SQL/PGQ extension has not been loaded");
+        }
+        auto sqlpgq_state = reinterpret_cast<SQLPGQContext *>(sqlpgq_state_entry->second.get());
+
+        auto csrID = ((CSRScanVData *)(data_p.bind_data))->csrID;
+        CSR *csr = sqlpgq_state -> GetCSR(csrID);
+        output.SetCardinality(csr->vsize);
+        output.data[0].SetVectorType(VectorType::FLAT_VECTOR);
+        FlatVector::SetData(output.data[0], (data_ptr_t)(reinterpret_cast<int64_t *>(csr->v)));
+}
+
+static void ScanCSRWFunction(ClientContext &context, TableFunctionInput &data_p, DataChunk &output) {
+        bool &gstate = ((CSRScanState &)*data_p.global_state).finished;
+
+        if(gstate) {
+                output.SetCardinality(0);
+                return;
+        }
+
+        gstate = true;
+
+        auto sqlpgq_state_entry = context.registered_state.find("sqlpgq");
+        if (sqlpgq_state_entry == context.registered_state.end()) {
+                //! Wondering how you can get here if the extension wasn't loaded, but leaving this check in anyways
+                throw InternalException("The SQL/PGQ extension has not been loaded");
+        }
+        auto sqlpgq_state = reinterpret_cast<SQLPGQContext *>(sqlpgq_state_entry->second.get());
+
+        auto csrID = ((CSRScanWData *)(data_p.bind_data))->csrID;
```

Comment:
please make naming consistent (See https://github.com/duckdb/duckdb/blob/master/CONTRIBUTING.md), so should be csr_id